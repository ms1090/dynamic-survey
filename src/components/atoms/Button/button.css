.button {
  background: #97c759;
  border: none;
  border-radius: 3px;
  box-sizing: border-box;
  color: white;
  cursor: pointer;
  font-weight: bold;
  margin-bottom: 10px;
  padding: 10px 20px;
  width: 100%;
}
.button:disabled {
  background: grey;
  cursor: default;
}

@media (min-width: 769px) {
  .button {
    margin-bottom: auto;
    width: auto;
  }
}
