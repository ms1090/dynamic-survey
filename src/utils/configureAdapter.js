import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
