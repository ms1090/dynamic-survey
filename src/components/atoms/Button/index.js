import React from 'react';
import PropTypes from 'prop-types';
import './button.css';

/**
 * Display a button as per customization props
 * @param {String} className CSS class name
 * @param {React.Element} children Button element children
 * @param {Boolean} primary Is button primary
 * @param {Boolean} secondary Is button secondary
 * @param {Function} onClick Click event handler function
 * @return {React.Element} Return Button component UI
 */
const Button = ({
  className,
  children,
  primary,
  secondary,
  onClick,
  ...others
}) => (
  <button
    className={`button ${className}${primary ? ' button-primary' : ''}${secondary ? ' button-secondary' : ''}`}
    onClick={onClick}
    type="button"
    {...others}
  >
    {children}
  </button>
);
// default values for component props
Button.defaultProps = {
  className: '',
  children: null,
  primary: false,
  secondary: false,
  onClick: null,
};
// prop types checking
Button.propTypes = {
  className: PropTypes.string,
  children: PropTypes.string || PropTypes.element,
  primary: PropTypes.bool,
  secondary: PropTypes.bool,
  onClick: PropTypes.func,
};
export default Button;
