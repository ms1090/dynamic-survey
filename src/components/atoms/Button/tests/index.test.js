import React from 'react';
import { shallow } from 'enzyme';
import Button from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('Button Component', () => {
  const props = {
    className: 'test-class',
    children: null,
    primary: false,
    secondary: false,
    onClick: null,
  };
  test('should render correctly', () => {
    let wrapper = shallow(<Button {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper = shallow(<Button {...props} primary />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper = shallow(<Button {...props} secondary />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
