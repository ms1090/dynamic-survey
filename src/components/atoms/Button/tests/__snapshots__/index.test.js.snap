// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
<button
  className="button test-class"
  onClick={null}
  type="button"
/>
`;

exports[`Button Component should render correctly 2`] = `
<button
  className="button test-class button-primary"
  onClick={null}
  type="button"
/>
`;

exports[`Button Component should render correctly 3`] = `
<button
  className="button test-class button-secondary"
  onClick={null}
  type="button"
/>
`;
